src/main/java/de/tum/in/www1/artemis/config/migration/entries/BambooMigrationService.java

```diff
@@ -0,0 +1,499 @@
+
+import de.tum.in.www1.artemis.domain.AuxiliaryRepository;
+import de.tum.in.www1.artemis.domain.VcsRepositoryUrl;
+import de.tum.in.www1.artemis.domain.participation.ProgrammingExerciseStudentParticipation;
+import de.tum.in.www1.artemis.repository.ProgrammingExerciseStudentParticipationRepository;
+import de.tum.in.www1.artemis.service.connectors.bamboo.BambooInternalUrlService;
+import de.tum.in.www1.artemis.service.connectors.bamboo.BambooService;
+
+/**
+ * Services for executing migration tasks for Bamboo.
+ */
+@Service
+@Profile("bamboo")
+public class BambooMigrationService implements CIVCSMigrationService {
+
+    private final RestTemplate restTemplate;
+
+    @Value("${artemis.continuous-integration.url}")
+    protected URL bambooServerUrl;
+
+    private final BambooInternalUrlService bambooInternalUrlService;
+
+    private final BambooService bambooService;
+
+    @Value("${server.url}")
+    private String artemisServerUrl;
+
+    @Value("${artemis.version-control.user}")
+    private String gitUser;
+
+    private Optional<Long> sharedCredentialId = Optional.empty();
+
+    private final Logger log = LoggerFactory.getLogger(BambooMigrationService.class);
+
+    public BambooMigrationService(@Qualifier("bambooRestTemplate") RestTemplate restTemplate, BambooService bambooService, BambooInternalUrlService bambooInternalUrlService) {
+        this.restTemplate = restTemplate;
+        this.bambooService = bambooService;
+        this.bambooInternalUrlService = bambooInternalUrlService;
+    }
+
+    /**
+     * Returns a list of all data-item-ids in the given html.
+     *
+     * @param html response that contains a list of items with the data-item-id attribute
+     * @return a list of all data-item-ids in the given html
+     */
+    private static List<Long> getDataItemIds(String html) {
+        if (html == null) {
+            return List.of();
+        }
+        Elements repositories = Jsoup.parse(html).select(".item");
+        List<Long> ids = new ArrayList<>();
+        for (Element repository : repositories) {
+            String repositoryIdString = repository.attr("data-item-id");
+            ids.add(Long.parseLong(repositoryIdString));
+        }
+        return ids;
+    }
+
+    /**
+     * Returns the name of the repository with the given id.
+     *
+     * @param html Response to the request to the editChainRepository.action endpoint
+     * @param id   The id of the repository we need the name for
+     * @return the name of the repository with the given id
+     */
+    private static Optional<String> getRepositoryNameById(String html, Long id) {
+        if (html == null) {
+            return Optional.empty();
+        }
+        Elements repositories = Jsoup.parse(html).select(".item");
+        for (Element repository : repositories) {
+            var link = Jsoup.parse(repository.html()).selectFirst("a");
+            if (link == null) {
+                continue;
+            }
```

Comment:
If we can't find a link, something went wrong - right? Should we throw here?

Reply:
If no link is found, there will be an info log (if the old repository is not found -> we add it, so no big deal), and if a new repository is not found that should be there, we log an error. Lines 157 and 216 in the same file, respectively.